bumpversion/cli.py

```diff
@@ -0,0 +1,653 @@
+
+    positionals, args = split_args_in_optional_and_positional(
+        sys.argv[1:] if original_args is None else original_args
+    )
+
+    if len(positionals[1:]) > 2:
+        warnings.warn(
+            "Giving multiple files on the command line will be deprecated, please use [bumpversion:file:...] in a config file.",
+            PendingDeprecationWarning,
+        )
+
+    parser1 = argparse.ArgumentParser(add_help=False)
+
+    parser1.add_argument(
+        "--config-file",
+        metavar="FILE",
+        default=argparse.SUPPRESS,
+        required=False,
+        help="Config file to read most of the variables from (default: .bumpversion.cfg)",
+    )
+
+    parser1.add_argument(
+        "--verbose",
+        action="count",
+        default=0,
+        help="Print verbose logging to stderr",
+        required=False,
+    )
+
+    parser1.add_argument(
+        "--list",
+        action="store_true",
+        default=False,
+        help="List machine readable information",
+        required=False,
+    )
+
+    parser1.add_argument(
+        "--allow-dirty",
+        action="store_true",
+        default=False,
+        help="Don't abort if working directory is dirty",
+        required=False,
+    )
+
+    known_args, remaining_argv = parser1.parse_known_args(args)
+
+    logformatter = logging.Formatter("%(message)s")
+
+    if len(logger_list.handlers) == 0:
+        ch2 = logging.StreamHandler(sys.stdout)
+        ch2.setFormatter(logformatter)
+        logger_list.addHandler(ch2)
+
+    if known_args.list:
+        logger_list.setLevel(1)
+
+    log_level = {0: logging.WARNING, 1: logging.INFO, 2: logging.DEBUG}.get(
+        known_args.verbose, logging.DEBUG
+    )
+    root_logger = logging.getLogger('')
+    root_logger.setLevel(log_level)
+
+    logger.debug("Starting {}".format(DESCRIPTION))
+
+    defaults = {}
+    vcs_info = {}
+
+    for vcs in VCS:
+        if vcs.is_usable():
+            vcs_info.update(vcs.latest_tag_info())
+
+    if "current_version" in vcs_info:
+        defaults["current_version"] = vcs_info["current_version"]
+
+    explicit_config = hasattr(known_args, "config_file")
+
+    if explicit_config:
+        config_file = known_args.config_file
+    elif not os.path.exists(".bumpversion.cfg") and os.path.exists("setup.cfg"):
+        config_file = "setup.cfg"
+    else:
+        config_file = ".bumpversion.cfg"
+
+    # setup.cfg supports interpolation - for compatibility we must do the same.
+    if os.path.basename(config_file) == "setup.cfg":
+        config = ConfigParser("")
+    else:
+        config = RawConfigParser("")
+
+    # don't transform keys to lowercase (which would be the default)
+    config.optionxform = lambda option: option
+
+    config.add_section("bumpversion")
+
+    config_file_exists = os.path.exists(config_file)
+
+    part_configs = {}
+
+    files = []
+
+    if config_file_exists:
+
+        logger.info("Reading config file {}:".format(config_file))
```

Comment:
The subsequent line definitely is. For this, since it's just a reorg and no code has changed (this is literally the existing code) I say we keep it and I'll address in a separate PR.